From 589d865391abca7ccc985bb38b2ecf50b8d7d1f1 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 21 Apr 2022 11:07:06 +0300
Subject: [PATCH] [SW-82718] rom: skip logic for slave die

Die id check is added in order to skip processing on slave die.

Change-Id: Ic9368186dba1a913a98abce41b965ec346ebd9e4
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 315da89..9fd94e2 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -375,6 +375,9 @@
 
 	security_rr_config();
 
+	if (arcboot_rom_global.die_id)
+		return;
+
 	/* store security relevant data in Scratch PAD Sram */
 	security_load_sec_efuse_dword();
 	security_load_revoke_bits();
diff --git a/common/boot_desc.c b/common/boot_desc.c
index c877966..d33f61f 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -62,6 +62,9 @@
 	uint32_t copy_sz;
 	static struct hl_fw_img_desc *p_img_desc;
 
+	if (arcboot_rom_global.die_id)
+		return 0;
+
 	if (img >= HL_FW_BOOT_PTR_MAX_IMG)
 		return 0;
 
